### PR TITLE
[SW-1878] Replace H2O_EXTENDED_JAR with H2O_DRIVER_JAR in all cases

### PR DESCRIPTION
**Pull request by mn-mikke**

### CODE REVIEW

File: doc/src/site/sphinx/deployment/backends.rst [hunk 1]
```diff
@@ -101,40 +101,13 @@ lead (as in the previous mode) to h2o cluster being killed as well.
...
- .. code:: bash
-
-    ./bin/get-extended-h2o.sh
-
-The script downloads the jar to the current directory and prints the absolute path to the downloaded jar.
+.. code:: bash
 
-**Note**: If you want to get an extended H2O jar for Sparkling Water and H2O versions that have not yet been
-released, you need to extend the JAR manually. This is explained in the following tutorial: :ref:`extend_jar_manually`.
+    H2O_DRIVER_JAR=/path/to/h2o-driver.jar
```
Comment (jakubhava) [inline doc/src/site/sphinx/deployment/backends.rst]: Should this section still mention how to obtain the JAR -> `./bin/get-h2o-driver.sh`?

File: doc/src/site/sphinx/deployment/backends.rst [hunk 2]
```diff
@@ -144,11 +117,11 @@ environment at the moment. We recommend this approach, as it is easier to deploy
 and it is also more suitable for production environments. When the H2O cluster is started on YARN, it is started
 as a map reduce job, and it always uses the flatfile approach for nodes to cloud up.
 
-Get extended H2O driver, for example, for cdh 5.8:
+Get H2O driver, for example, for cdh 5.8:
 
 .. code:: bash
 
-    H2O_EXTENDED_JAR=$(./bin/get-extended-h2o.sh cdh5.8)
+    H2O_DRIVER_JAR=$(./bin/get-h2o-driver.sh cdh5.8 | tail -n 1 | awk 'NF>1{print $NF})
```
Comment (jakubhava) [inline doc/src/site/sphinx/deployment/backends.rst]: Is this part necessary? `| tail -n 1 | awk 'NF>1{print $NF})` I would prefer fixing the `./bin/get-h2o-driver.sh` and keep the user-facing command simple, WDYT?

File: doc/src/site/sphinx/deployment/backends.rst (hunk 1 above)
Comment (mn-mikke) [reply doc/src/site/sphinx/deployment/backends.rst]: Good idea, will change it!

File: doc/src/site/sphinx/deployment/backends.rst (hunk 2 above)
Comment (mn-mikke) [reply doc/src/site/sphinx/deployment/backends.rst]: Np, will do.

**Comment by jakubhava** [comment]
@mn-mikke one last place we need to change a little bit of doc -> in dist/src/index.html (our download page), in the section `h2>Kluster</h2>`

**Comment by mn-mikke** [comment]
@jakubhava will do in a followup PR if you don't mind...

**Comment by jakubhava** [comment]
yup, no problem, my bad I realized it just now :)

What do you think to change that doc on the download page to use automatic cluster start instead? And we can point people to documentation for info on manual start.

**Pull request merged**